nym-vpn-core/crates/nym-vpn-lib-types/src/tunnel_state.rs

```diff
@@ -198,6 +205,117 @@ pub enum ErrorStateReason {
     Internal(String),
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ClientErrorReason {
+    Firewall,
+    Routing,
+    SameEntryAndExitGateway,
+    InvalidEntryGatewayCountry,
+    InvalidExitGatewayCountry,
+    MaxDevicesReached,
+    BandwidthExceeded,
+    SubscriptionExpired,
+    Dns(String),
+    Api(String),
+    Internal(String),
+}
+
+impl From<ErrorStateReason> for ClientErrorReason {
+    fn from(value: ErrorStateReason) -> Self {
+        match value {
+            ErrorStateReason::SameEntryAndExitGateway => Self::SameEntryAndExitGateway,
+            ErrorStateReason::InvalidEntryGatewayCountry => Self::InvalidEntryGatewayCountry,
+            ErrorStateReason::InvalidExitGatewayCountry => Self::InvalidExitGatewayCountry,
+            ErrorStateReason::BadBandwidthIncrease => Self::Api(value.as_ref().to_string()),
+            ErrorStateReason::SyncAccount(err) => err.into(),
+            ErrorStateReason::SyncDevice(err) => err.into(),
+            ErrorStateReason::RegisterDevice(err) => err.into(),
+            ErrorStateReason::RequestZkNym(err) => err.into(),
+            ErrorStateReason::RequestZkNymBundle {
+                successes: _,
+                failed,
+            } => {
+                // Return the first error if it exists, otherwise return a default error
+                if let Some(first_error) = failed.first() {
+                    ClientErrorReason::from(first_error.clone())
+                } else {
+                    Self::Api("Empty failure list in RequestZkNymBundle".to_string())
+                }
+            }
+            ErrorStateReason::Firewall => Self::Firewall,
+            ErrorStateReason::TunDevice
+            | ErrorStateReason::TunnelProvider
+            | ErrorStateReason::DuplicateTunFd => Self::Internal(value.as_ref().to_string()),
+            ErrorStateReason::Internal(message) => Self::Internal(message),
+            ErrorStateReason::Routing => Self::Routing,
+            ErrorStateReason::ResolveGatewayAddrs => Self::Dns(value.as_ref().to_string()),
+            ErrorStateReason::StartLocalDnsResolver => Self::Dns(value.as_ref().to_string()),
+            ErrorStateReason::Dns => Self::Dns(value.as_ref().to_string()),
```

Comment:
Since we use string for the inner values, better to implement `strum::Display` for `ErrorStateReason` to avoid those refs to string conversion